src/main/java/echo/parser/Parser.java

```diff
@@ -0,0 +1,256 @@
+        return new ListCommand();
+    }
+
+    /**
+     * Prepare list command.
+     *
+     * @param desc Description of todo command.
+     * @return TodoCommand.
+     */
+    private static Command prepareTodoCommand(String desc) {
+        return new TodoCommand(desc);
+    }
+
+    /**
+     * Prepare deadline command.
+     *
+     * @param input Input from user.
+     * @param desc Description of deadline command.
+     * @param localDateTime Local date and time.
+     *
+     * @return DeadlineCommand.
+     *
+     * @throws EchoException If input is invalid.
+     */
+    private static Command prepareDeadlineCommand(String input, String desc, LocalDateTime localDateTime)
+            throws EchoException {
+        if (!input.contains("/by")) {
+            // If /by is not specified.
+            throw new EchoException("Invalid input. Make sure to include '/by'\n"
+                    + "    E.g. deadline return book /by 2019-10-15 1800");
+        }
+        return new DeadlineCommand(desc, localDateTime);
+    }
+
+    /**
+     * Prepare event command.
+     *
+     * @param input Input from user.
+     * @param desc Description of event command.
+     * @param localDateTime Local date and time.
+     *
+     * @return EventCommand.
+     *
+     * @throws EchoException If input is invalid.
+     */
+    private static Command prepareEventCommand(String input, String desc, LocalDateTime localDateTime)
+            throws EchoException {
+        if (!input.contains("/at")) {
+            // If /at is not specified.
+            throw new EchoException("Invalid input. Make sure to include '/at' \n."
+                    + "    E.g. event meeting /at 2019-10-15 1800");
+        }
+        return new EventCommand(desc, localDateTime);
+    }
+
+    /**
+     * Prepare mark command.
+     *
+     * @param i Index of task.
+     *
+     * @return MarkCommand.
+     */
+    private static Command prepareMarkCommand(int i) {
+        return new MarkCommand(i);
+    }
+
+    /**
+     * Prepare unmark command.
+     *
+     * @param i Index of task.
+     *
+     * @return UnmarkCommand.
+     */
+    private static Command prepareUnmarkCommand(int i) {
+        return new UnmarkCommand(i);
+    }
+
+    /**
+     * Prepare delete command.
+     *
+     * @param i Index of task.
+     *
+     * @return DeleteCommand.
+     */
+    private static Command prepareDeleteCommand(int i) {
+        return new DeleteCommand(i);
+    }
+
+    /**
+     * Prepare find command.
+     *
+     * @param desc Description to find.
+     * @return FindCommand.
+     */
+    private static Command prepareFindCommand(String desc) {
+        return new FindCommand(desc);
+    }
+
+    /**
+     * Prepare help command.
+     *
+     * @return HelpCommand.
+     */
+    private static Command prepareHelpCommand() {
+        return new HelpCommand();
+    }
+
+    /**
+     * Prepare bye command.
+     *
+     * @return ByeCommand.
+     */
```

Comment:
I like how you wrote java doc for all classes and methods, and even for private methods! Good effort! 💯